packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx

```diff
@@ -91,9 +92,22 @@ class UtilityMenuItem extends React.Component {
     }
   }
 
+  handleIsSelectable() {
+    if (this.props.isSelectable) {
+      this.setState({ isSelected: !this.state.isSelected });
+    }
+  }
+
+  handleSelection(event, key) {
+    event.preventDefault();
+    this.handleIsSelectable();
+    this.props.onChange(event, key);
+  }
+
   wrapOnKeyDown(key, onKeyDown) {
     return ((event) => {
       if (event.nativeEvent.keyCode === Utils.KEY_CODES.ENTER || event.nativeEvent.keyCode === Utils.KEY_CODES.SPACE || event.nativeEvent.keyCode === Utils.KEY_CODES.RIGHT_ARROW) {
+        this.handleIsSelectable();
         this.props.onChange(event, key);
```

Comment:
Both `wrapOnKeyDown` and `handleSelection` call `handleIsSelectable` & `onChange`. Could we move the `handleIsSelectable` call to be within the`onChange`?

Reply:
WHOOPS. JUST KIDDING 😅 its `this.props.onChange`